tables_app/src/main/java/org/opendatakit/tables/activities/MainActivity.java

```diff
@@ -299,7 +301,21 @@ public void initializationCompleted() {
 
   @Override
   public void onBackPressed() {
-    popBackStack();
+    new AlertDialog.Builder(this)
+            .setIcon(android.R.drawable.ic_dialog_alert)
+            .setTitle("Closing Application")
+            .setMessage("Are you sure you want to close this application?")
+            .setPositiveButton("Yes", new DialogInterface.OnClickListener()
+            {
+              @Override
+              public void onClick(DialogInterface dialog, int which) {
+                finish();
+              }
+
+            })
+            .setNegativeButton("No", null)
+            .show();
+    //popBackStack();
```

Comment:
I do not believe this solution will work. As the back stack tracks the screens that have been rendered in the WebUI. By commenting out popBackStack() you are changing the way Tables is designed to operate.

The warning should only happen when there is nothing left on the back stack. Otherwise it should bring up the other back stack entry. Check out line 265 as that is the case when finish is getting called because there is nothing left on the backStack